sonic_installer/main.py

```diff
@@ -0,0 +1,165 @@
+#! /usr/bin/python -u
+
+import sys
+import click
+import urllib
+import subprocess
+
+MACHINE_PATH = '/host'
+IMAGE_PREFIX = 'SONiC-OS'
+IMAGE_DIR_PREFIX = 'image'
+DEFAULT_IMAGE_PATH = '/tmp/sonic_installer'
+
+
+# Run bash command and print output to stdout
+def run_command(command, pager=False):
+    click.echo(click.style("Command: ", fg='cyan') + click.style(command, fg='green'))
+    p = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+    if pager is True:
+        click.echo_via_pager(p.stdout.read())
+    else:
+        click.echo(p.stdout.read())
+    p.wait()
+    if p.returncode != 0:
+        sys.exit(p.returncode)
+
+
+# Returns list of installed images
+def get_installed_images():
+    images = []
+    config = open(MACHINE_PATH + '/grub/grub.cfg', 'r')
+    for line in config:
+        if line.startswith('menuentry'):
+            image = line.split()[1].strip("'")
+            if image != 'ONIE':
+                images.append(image)
+    config.close()
+    return images
+
+
+# Returns name of current image
+def get_current_image():
+    images = get_installed_images()
+    current = ''
+    cmdline = open('/proc/cmdline', 'r')
+    for line in cmdline:
+        for image in images:
+            image_dir = image.replace(IMAGE_PREFIX, IMAGE_DIR_PREFIX)
+            if line.find(image_dir):
+                current = image
+                break
+    cmdline.close()
+    return current
+
+
+# Callback for confirmation prompt. Aborts if user enters "n"
+def abort_if_false(ctx, param, value):
+    if not value:
+        ctx.abort()
+
+
+# Main entrypoint
+@click.group()
+def cli():
+    """ SONiC image installation manager """
+    pass
+
+
+# Install image
+@cli.command()
+@click.option('-y', '--yes', is_flag=True, callback=abort_if_false,
+        expose_value=False, prompt='New image will be installed, continue?')
+@click.argument('url')
+def install(url):
+    """ Install image from local binary or URL"""
+    cleanup_image = False
+    if url.startswith('http://') or url.startswith('https://'):
+        click.echo('Downloading image...')
+        urllib.urlretrieve(url, DEFAULT_IMAGE_PATH)
+        image_path = DEFAULT_IMAGE_PATH
+    else:
+        image_path = url
+
+    run_command("sh " + image_path)
```

Comment:
>run_command [](start = 4, length = 11)

Can we remove the explicit "sh"? The image should be executable and the shebang could be arbitrary.